src/components/StarRating/src/StarRating.vue

```diff
@@ -0,0 +1,157 @@
+<template>
+	<div :style="starComputedStyles">
+		<star
+			v-for="star in STAR_COUNT"
+			:key="star"
+			:fill="fillForStar(star)"
+			:color="color"
+			:class="$s.Star"
+			@hover="hoverStar(star)"
+			@unhover="unhoverStar(star)"
+			@click="clickStar(star)"
+		/>
+	</div>
+</template>
+
+<script>
+import chroma from 'chroma-js';
+import Star from './Star.vue';
+
+const MIN_RATING = 0;
+const MAX_RATING = 5;
+const HALF_RATING = 0.5;
+// eslint-disable-next-line no-magic-numbers
+const STAR_COUNT = 5;
+
+const STAR_STYLES = {
+	small: {
+		height: 16,
+		spacing: 1,
+	},
+	medium: {
+		height: 24,
+		spacing: 2,
+	},
+	large: {
+		height: 32,
+		spacing: 4,
+	},
+};
+
+export default {
+	name: 'StarRating',
+
+	components: {
+		Star,
+	},
+
+	props: {
+		/**
+		 * Rating value from 0-5, determines fill state of stars
+		 */
+		rating: {
+			type: Number,
+			default: 0,
+			validate: (rating) => rating >= MIN_RATING && rating <= MAX_RATING,
+		},
+
+		/**
+		 * Size of rating component
+		 */
+		size: {
+			type: String,
+			default: 'medium',
+			validate: (size) => ['small', 'medium', 'large'].includes(size),
+		},
+		/**
+		 * Color of the star
+		 */
+		color: {
+			type: String,
+			default: '#FFBF00',
+			validator: (color) => chroma.valid(color),
+		},
+
+		/**
+		 * Determines whether to bubble up click/hover events and show pointer cursor
+		 */
+		editable: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	data() {
+		return {
+			STAR_COUNT,
+			hoveredRating: undefined,
```

Comment:
Why would we initiate this vaule as `undefined` here?

Reply:
Not sure what else it would be initialized as. This value isn't set until the user hover over one of the stars.

Reply:
To me, `0` or `null` would make more sense here since that is more in line with a numerical value. It's still falsey so your code will evaluate the same.

Reply:
Fair enough. Though it seems that `null` isn't allowed with the eslint config here.
![image](https://user-images.githubusercontent.com/3402466/134530566-c3e0ce32-3513-4232-b279-9e6bf1716aa0.png)

Reply:
@landondurnan @pretzelhammer @privatenumber is there a reason why we decided to enforce the usage of `null` in our linting in Maker?

Reply:
Our eslint rules: https://github.com/square/maker/blob/master/.eslintrc.js

The unicorn/no-null recommendations with articles on "Why" here: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-null.md

But always love hearing Crockford give a reason: https://www.youtube.com/watch?v=PSGEjv3Tqo0&t=563s

Reply:
I had honestly never thought of that re: Crockford's comments :mindblown: